src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -3040,6 +3012,48 @@ void MergeTreeData::checkAlterIsPossible(const AlterCommands & commands, Context
     columns_in_keys.insert(columns_alter_type_forbidden.begin(), columns_alter_type_forbidden.end());
     columns_in_keys.insert(columns_alter_type_metadata_only.begin(), columns_alter_type_metadata_only.end());
     columns_in_keys.insert(columns_alter_type_check_safe_for_partition.begin(), columns_alter_type_check_safe_for_partition.end());
+    return columns_in_keys;
+}
+
+NameSet MergeTreeData::getAllKeyColumns() const
+{
+    NameSet columns_alter_type_forbidden;
+    NameSet columns_alter_type_metadata_only;
+    NameSet columns_alter_type_check_safe_for_partition;
+    return getAllKeyColumns(columns_alter_type_forbidden, columns_alter_type_metadata_only, columns_alter_type_check_safe_for_partition);
+}
+
+void MergeTreeData::checkAlterIsPossible(const AlterCommands & commands, ContextPtr local_context) const
+{
+    /// Check that needed transformations can be applied to the list of columns without considering type conversions.
+    StorageInMemoryMetadata new_metadata = getInMemoryMetadata();
+    StorageInMemoryMetadata old_metadata = getInMemoryMetadata();
+
+    const auto & settings = local_context->getSettingsRef();
+    const auto & settings_from_storage = getSettings();
+
+    if (!settings.allow_non_metadata_alters)
+    {
+        auto mutation_commands = commands.getMutationCommands(new_metadata, settings.materialize_ttl_after_modify, local_context);
+
+        if (!mutation_commands.empty())
+            throw Exception(ErrorCodes::ALTER_OF_COLUMN_IS_FORBIDDEN,
+                            "The following alter commands: '{}' will modify data on disk, "
+                            "but setting `allow_non_metadata_alters` is disabled",
+                            queryToString(mutation_commands.ast()));
+    }
+
+    commands.apply(new_metadata, local_context);
+
+    if (commands.hasInvertedIndex(new_metadata) && !settings.allow_experimental_inverted_index)
+        throw Exception(ErrorCodes::SUPPORT_IS_DISABLED,
+                "Experimental Inverted Index feature is not enabled (turn on setting 'allow_experimental_inverted_index')");
+
```

Comment:
This change related only to extraction of code from `MergeTreeData::checkAlterIsPossible` to `MergeTreeData::getAllKeyColumns`